framework/Assembler.py

```diff
@@ -39,13 +39,14 @@ def __init__(self):
       @ In, None
       @ Out, None
     """
-    self.type               = self.__class__.__name__  # type
-    self.name               = self.__class__.__name__  # name
-    self.assemblerObjects   = {}                       # {MainClassName(e.g.Distributions):[class(e.g.Models),type(e.g.ROM),objectName]}
+    self.type = self.__class__.__name__  # type
+    self.name = self.__class__.__name__  # name
+    if not hasattr(self, 'assemblerObjects'): # protect against polyinheritance woes
```

Comment:
That's a separate issue; this is protecting against the constructor emptying out the assemblerObjects member; the multiple target evaluation is because of multiple eligible output DataObjects that could be TargetEvaluation objects in the Step.